hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DBCheckpointServlet.java

```diff
@@ -123,6 +133,24 @@ private boolean hasPermission(UserGroupInformation user) {
     }
   }
 
+  private void deleteContents(Path directory) throws IOException {
+    Files.walkFileTree(directory, new SimpleFileVisitor<Path>() {
+      @Override
+      public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+        Files.delete(file); // Delete file
+        return FileVisitResult.CONTINUE;
+      }
+
+      @Override
+      public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+        if (!dir.equals(directory)) { // Skip deleting the root directory
+          Files.delete(dir);
+        }
+        return FileVisitResult.CONTINUE;
+      }
+    });
+  }
```

Comment:
Can we use any existing utility for the same?

- [Apache Commons IO](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html#cleanDirectory(java.io.File))
- [Hadoop](https://hadoop.apache.org/docs/r3.4.1/api/org/apache/hadoop/fs/FileUtil.html#fullyDeleteContents-java.io.File-)

Reply:
+1